### PR TITLE
Improve error handling in some places

### PR DESCRIPTION
**Pull request by NebelNidas**
Extracted from https://github.com/QuiltMC/enigma/commit/bad1042912d31a00a455c5e7931c95b68e7307b4.

### CODE REVIEW

File: enigma/src/main/java/cuchaz/enigma/analysis/index/JarIndex.java [hunk 1]
```diff
@@ -84,7 +84,12 @@ public void indexJar(Set<String> classNames, ClassProvider classProvider, Progre
 		progress.step(2, I18n.translate("progress.jar.indexing.references"));
 
 		for (String className : classNames) {
-			classProvider.get(className).accept(new IndexReferenceVisitor(this, entryIndex, inheritanceIndex, Enigma.ASM_VERSION));
+			try {
+				classProvider.get(className).accept(new IndexReferenceVisitor(this, entryIndex, inheritanceIndex, Enigma.ASM_VERSION));
+			} catch (Exception e) {
+				System.err.println("Exception while indexing class: " + className + ":");
+				e.printStackTrace();
```
Comment (modmuss50) [inline enigma/src/main/java/cuchaz/enigma/analysis/index/JarIndex.java:91]: This should throw an new RuntimeException with a message containing the class name.

We dont want to contiune when in a bad state.

File: enigma-cli/src/main/java/cuchaz/enigma/command/Main.java [hunk 2]
```diff
@@ -48,13 +48,13 @@ public static void main(String... args) throws Exception {
 				throw new CommandHelpException(cmd, ex);
 			}
 		} catch (CommandHelpException ex) {
-			System.err.println(ex.getMessage());
+			ex.printStackTrace();
```
Comment (modmuss50) [inline enigma-cli/src/main/java/cuchaz/enigma/command/Main.java:51]: Is the stacktrace useful here? I dont think so.

File: enigma-cli/src/main/java/cuchaz/enigma/command/Main.java [hunk 3]
```diff
@@ -48,13 +48,13 @@ public static void main(String... args) throws Exception {
...
 			System.out.println(String.format("%s - %s", Enigma.NAME, Enigma.VERSION));
 			System.out.println("Command " + ex.command.name + " has encountered an error! Usage:");
 			printHelp(ex.command);
 			System.exit(1);
 		} catch (IllegalArgumentException ex) {
-			System.err.println(ex.getMessage());
+			ex.printStackTrace();
```
Comment (modmuss50) [inline enigma-cli/src/main/java/cuchaz/enigma/command/Main.java:57]: Same as above.

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java [hunk 4]
```diff
@@ -12,23 +12,27 @@
...
+	}
+
+	public MappingParseException(String filename, int line, String message) {
 		this.line = line;
 		this.message = message;
-		filePath = file.getAbsolutePath();
+		filePath = filename;
 	}
 
-	public MappingParseException(Supplier<String> filenameProvider, int line, String message) {
```
Comment (modmuss50) [inline enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java:28]: Would it not be better to just take the `Path` instead?

**Review by modmuss50 — CHANGES_REQUESTED** [review]

**Comment by modmuss50** [comment]
The build also failed due to the checkstyle rules 👍

File: enigma-cli/src/main/java/cuchaz/enigma/command/Main.java (hunk 2 above)
Comment (NebelNidas) [reply enigma-cli/src/main/java/cuchaz/enigma/command/Main.java:51]: Yeah, would clutter the output more than it'd help, I'm going to revert the change

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java (hunk 4 above)
Comment (NebelNidas) [reply enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java:28]: Good suggestion!

File: enigma/src/main/java/cuchaz/enigma/analysis/index/JarIndex.java [hunk 5]
```diff
@@ -84,7 +84,11 @@ public void indexJar(Set<String> classNames, ClassProvider classProvider, Progre
 		progress.step(2, I18n.translate("progress.jar.indexing.references"));
 
 		for (String className : classNames) {
-			classProvider.get(className).accept(new IndexReferenceVisitor(this, entryIndex, inheritanceIndex, Enigma.ASM_VERSION));
+			try {
+				classProvider.get(className).accept(new IndexReferenceVisitor(this, entryIndex, inheritanceIndex, Enigma.ASM_VERSION));
+			} catch (Exception e) {
+				throw new RuntimeException("Exception while indexing class '" + className + "'':\n" + e);
```
Comment (modmuss50) [inline enigma/src/main/java/cuchaz/enigma/analysis/index/JarIndex.java:90]:
```suggestion
				throw new RuntimeException("Exception while indexing class: " + className,  e);
```

**Review by modmuss50 — CHANGES_REQUESTED** [review]

File: enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java [hunk 6]
```diff
@@ -12,27 +12,32 @@
...
-		this.message = message;
-		filePath = filenameProvider.get();
+		this.message = cause.toString();
+		this.filePath = filePath;
 	}
 
 	@Override
 	public String getMessage() {
-		return "Line " + line + ": " + message + " in file " + filePath;
+		return "Line " + line + ": " + message + " in file " + filePath.toAbsolutePath().toString();
```
Comment (liach) [inline enigma/src/main/java/cuchaz/enigma/translation/mapping/serde/MappingParseException.java:41]: Absolute paths may be too long if the working directory is too nested; I think the old `filePath.toString` should suffice

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**